src/components/SideBar.tsx

```diff
@@ -42,21 +53,35 @@ function AddCourse() {
 	console.log('course added');
 }
 
-function CourseList({ courses }: { courses: Course[] }) {
+function CourseList({ courses, activeCourse, onChange }: CourseListInterface) {
+	console.log('active course is: ' + activeCourse.name);
 	return (
 		<ListOfCoursesContainer>
 			{courses.map((item, name) => {
-				return <CourseItem key={name}>{item.name}</CourseItem>;
+				return (
+					// need to check if the item is the active course item or not
+					<CourseItem key={name} onClick={() => onChange(item)}>
+						{item.name}
+					</CourseItem>
+				);
 			})}
 		</ListOfCoursesContainer>
 	);
 }
-function SideBar({ currentSemester, ...props }: SideBarInterface) {
+function SideBar({
+	currentSemester,
+	activeCourse,
+	onChange,
+}: SideBarInterface) {
 	return (
-		<SideBarContainer {...props}>
+		<SideBarContainer>
```

Comment:
Make sure `{...props}` is here so we can style this with styled-components!

Reply:
fixed :)